docs(buildd): Document presence and use of buildd images (#354)
canonical/ubuntu-cloud-docs: public-images/public-images-explanation/buildd.rst

> Ubuntu Buildd Images

Reviewer: ```suggestion
Ubuntu Buildd images
```
We use [sentence case for our headings](https://docs.ubuntu.com/styleguide/en#headings)

====================

Buildd images are another offering of cloud images built by Canonical that are used primarly to serve the Launchpad `Build Farm <https://documentation.ubuntu.com/launchpad/en/latest/reference/services/build-farm/#>`_.
These images are publicly available to download and use and can be found under `cloud-images.ubuntu.com/buildd <https://cloud-images.ubuntu.com/buildd/>`_. 

> Ubuntu Buildd Images Use Cases

Reviewer: ```suggestion
Ubuntu Buildd image use cases
```

------------------------------
Buildd images are intended to be used for building Ubuntu packages or other Ubuntu images as detailed in the `Build Farm <https://documentation.ubuntu.com/launchpad/en/latest/reference/services/build-farm/#detailed-description>`_ description section:

* Deb packages for the `Ubuntu Archive <https://archive.ubuntu.com/>`_
* Snaps for Canonical's `Snapcraft <https://snapcraft.io/>`_
* Charms for Canonical's `Charmhub <https://charmhub.io/>`_
* Ubuntu images based on the `Ubuntu Base <https://wiki.ubuntu.com/Base>`_ project.

> Difference from other Cloud Images

Reviewer: ```suggestion
Difference from other cloud images
```

----------------------------------

Buildd images are meant to be light and very minimal compared to other cloud images offered by Canonical. This is due to the following:

* They contain only a small set of packages that are required to build new packages (such as debs or snaps), charms, or images.
* Direct access to the internet is not enabled for these images; it can be granted for a restricted set of URLs through token authentication.

> * The images published as a combined lxd tarballs are not bootable and do not contain a kernel unlike the QCoW disk images.

Reviewer: ```suggestion
* The images published as combined lxd tarballs are not bootable and do not contain a kernel unlike the QCoW disk images.
```
What are 'combined lxd tarballs'? The combined part is not clear.

Author: Perhaps this is jargon that should not be used in documentation. It is meant to imply a tarball that also packages the metadata with the image artefact. Let me know if there is any need to add context for it, or else use simpler phrasing.

Reviewer: Maybe we could just add that detail in "The images published as combined lxd tarballs (containing both the image artefact and its metadat) are not bootable .."

Reviewer: If that seems okay, I can add it in.

Author: Yes that works! It will avoid confusion. Thanks!

> Due to the lack of common userspace tools and additional configuration required to set up networking and internet access for Buildd images, Canonical does not recommend these images outside the use cases explained above. 

Reviewer: ```suggestion
Due to the lack of common user space tools and the additional configuration required to set up networking and internet access for Buildd images, Canonical does not recommend their use outside the use cases explained above. 
```
